src/hotspot/share/oops/symbol.cpp

```diff
@@ -294,6 +294,18 @@ void Symbol::print_as_signature_external_parameters(outputStream *os) {
   }
 }
 
+void Symbol::print_signature_as_external_field_type(outputStream *os) {
+  for (SignatureStream ss(this, false); !ss.is_done(); ss.next()) {
+    if (ss.is_array()) {
+      print_array(os, ss);
+    } else if (ss.is_reference()) {
+      print_class(os, ss);
+    } else {
+      os->print("%s", type2name(ss.type()));
+    }
+  }
+}
+
```

Comment:
The SignatureStream for field refs should have exactly one element, and must not be a return type, so this function should be tightened (so it will assert on inputs such as `(I)V` or  `()L/java/lang.Object;`, 
 
```
assert(!Signature::is_method(this), "must be a field signature");
SignatureStream ss(this, false); 
assert(!ss.is_done(), "must have at least one element in field ref");
assert(!at_return_type(), "field ref can not be a return type");

if (ss.is_array()) {
  print_array(os, ss);
} else if (ss.is_reference()) {
  print_class(os, ss);
} else {
  os->print("%s", type2name(ss.type()));
}

#ifdef ASSERT
ss.next();
assert(ss.is_done(), "must have at most one element in field ref");
#endif
```